### PR TITLE
feat: launch AVD

### PR DESCRIPTION
**Pull request by itsspriyansh**
This PR adds functionality for launching an AVD. The workflow is triggered using the `--emulator` flag with the `connect` subcomamnd as shown below:

`npx @nightwatch/mobile-helper android connect --emulator`

[Screencast from 04-07-24 10:39:32 PM IST.webm](https://github.com/nightwatchjs/mobile-helper-tool/assets/88396544/493be311-a311-4dbd-a4dc-3b5cfee57dc5)

### CODE REVIEW

**Comment by garg3133** [comment]
I think instead of using `--avd` we should use `--emulator` flag to trigger this workflow since we are actually connecting to an emulator environment and then choosing which AVD to run in that emulator. Also, "emulator" would be easier for users to remember and use when compared to "avd".

We could probably use `--avd` as a valued flag here using which users can pass the AVD name to run in the emulator with the command itself:
```sh
npx @nightwatch/mobile-helper android connect --emulator --avd nightwatch-android-11

# could also use the @ patter that the emulator binary provides
npx @nightwatch/mobile-helper android connect --emulator @nightwatch-android-11
```

**Comment by itsspriyansh** [comment]
@garg3133 I am updated the PR. Here I have used `as` keyword for valued option because we know that any boolean option will be handled by the `verifyOptions` function.

**Comment by itsspriyansh** [comment]
In this PR, `--avd` flag is used to specify the name of the avd to launch. If user runs the following command then the functionality works fine.
```
npx @nightwatch/mobile-helper android connect --emulator --avd <avd_name>
```
but if user passes a command like the one below
```
npx @nightwatch/mobile-helper android connect --emulator --avd
```
an error will be shown since `avd` is passed here as a boolean option and not a valued option. And the `verifyOptions` function in this PR #51 throws error for any unrecognized boolean flag. Since `avd` is not defined in the `AVAILABLE_SUBCOMMANDS` object it will be considered as unknown.

**Comment by garg3133** [comment]
> Even though this is incorrect usage to just pass the avd option without a value, I think we shouldn't show an error and just proceed with the script and prompt the user to select a device.

You are correct here. If the user passes `--avd` only, we should proceed with the normal flow instead of throwing an error.

One question that I have here is - why are we only checking for boolean flags in `verifyOptions` and not all the possible flags? Why not add `--avd` to `AVAILABLE_SUBCOMMANDS` as well and verify it? We would anyway need it there to show in the help section, won't we?

**Comment by garg3133** [comment]
In the `verifyOptions`, we just want to skip the flags that are set to `false` (which would mean that those flag wasn't actually passed with the command), otherwise we should verify all other flags.

**Comment by garg3133** [comment]
@itsspriyansh Can you see what is remaining here?

**Comment by itsspriyansh** [comment]
@garg3133 Sure. I have updated the PR.

File: src/commands/android/subcommands/connect/emulator.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      });
+      options.avd = avdAnswer.avdName;
+    }
+
+    Logger.log();
+
+    const connectionStatus = await launchAVD(sdkRoot, platform, options.avd as string);
+    if (!connectionStatus) {
+      return false;
+    }
```
Comment (garg3133) [inline src/commands/android/subcommands/connect/emulator.ts:55]: On doing this, the process gets stuck while the emulator is running. We should instead exit the process and let the emulator run in the background. Then, we can use the `disconnect` command to close the emulator.

Comment (garg3133) [reply src/commands/android/subcommands/connect/emulator.ts:55]: Could probably use the spawnSync function here... we don't need to kill the already running AVD in this case or wait for the AVD to boot up, so using the `launchAVD` here does not make sense.

**Comment by itsspriyansh** [comment]
@garg3133 PR updated. Using `spawnSync` for now but we will replace it with `spawnCommandSync` once #57 is merged.

**Comment by garg3133** [comment]
@itsspriyansh Since you were asking whether we should test subcommand scripts or not, the fix I did in the last commit of this PR would have been caught if we have tests because in those tests we would have tried passing different versions of `--avd` flag and it wouldn't have worked correctly for `boolean` or `array` avds.

**Pull request merged**